### DISCUSSION TITLE
Question on using docker-volume-backup with multiple docker compose files

### DISCUSSION THREAD

**Discussion by Frankst2** [offen/docker-volume-backup#294] [General]
Hi all,
I just started using offen/docker-volume-backup for local backups and it seems to work very well! However, I still have doubts about one thing: What is the correct way when I have multiple docker compose files/applications running on one server? Is it correct to insert the backup service (like described at https://github.com/offen/docker-volume-backup/blob/main/README.md) in every compose file I have? Will it automatically synchronize the backup so that it won't run the backup n times for every volume (n=number of docker compose files)?
When I tested it yesterday it looked like it ran the backup n times, but maybe it was because I was also testing manual backup runs :)

Regards
Frank

**Comment by m90** [comment]
Usually people use one of two approaches:

- one container off this image that is exclusively backing up a single service
- one container that is backing up multiple services at once

Both approaches work, so you can pick one depending on your needs.

There are quite a few issues discussing this already, e.g:
- https://github.com/offen/docker-volume-backup/issues/123#issuecomment-1178168541
- https://github.com/offen/docker-volume-backup/issues/29

There is also dedicated documentation on using multiple configuration files for a single container: https://offen.github.io/docker-volume-backup/how-tos/run-multiple-schedules.html

**Reply by Frankst2** [reply]
Thanks for the quick reply! I will look into this.

**Reply by Frankst2** [reply]
Another thing I noticed now: I've now tried the one-container-per-service approach. When I run the backup command manually for one service, it seems to run all pre-commands for all compose files:

```
time=2023-10-29T11:04:33.751Z level=INFO msg="Running docker-volume-backup.archive-pre command /bin/sh -c \"mysqldump [...]' --all-databases > /tmp/dumps/dump.sql\" for container mariadb-mariadb-1"
time=2023-10-29T11:04:33.751Z level=INFO msg="Running docker-volume-backup.archive-pre command /bin/sh -c \"pg_dumpall -U postgres > /tmp/dumps/dump_postgres.sql\" for container postgres-postgres-1"

```

Is there a way to avoid this?

**Reply by m90** [reply]
You can use `EXEC_LABEL` as documented here https://offen.github.io/docker-volume-backup/how-tos/run-custom-commands.html

**Comment by Frankst2** [comment]
Thanks
